### PR TITLE
Refactor StateAwareCategoryService::getPersonalisationResult() not to return null

### LINKED ISSUE #138: Refactor StateAwareCategoryService::getPersonalisationResult() not to return null
**Is your feature request related to a problem? Please describe.**
Currently `StateAwareCategoryService::getPersonalisationResult()` might return null, which leads to a lot of null checks across the code base.

**Describe the solution you'd like**
Refactor the class to not return null.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by dairbuirabass**
Refactor StateAwareCategoryService::getPersonalisationResult() not to return null

## Description
Remove null checks and init StateAwareCategoryService::lastResult to be of type CategoryMerchandisingResult even if Nosto Sorting is not used.

## Related Issue
Closes #138

## Motivation and Context
Remove extra if checks

## How Has This Been Tested?
Locally.

## Checklist:
- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [ ] I have assigned the correct milestone or created one if non-existent.
- [x] I have correctly labeled this pull request.
- [x] I have linked the corresponding issue in this description.
- [x] I have updated the corresponding Jira ticket.
- [x] I have requested a review from at least 2 reviewers
- [x] I have checked the base branch of this pull request
- [x] I have checked my code for any possible security vulnerabilities

### CODE REVIEW

File: Model/Service/Recommendation/StateAwareCategoryService.php [hunk 1]
```diff
@@ -247,7 +247,7 @@ function () use ($nostoAccount, $previewMode, $category, $pageNumber, $limit, $f
     /**
      * @inheritDoc
      */
-    public function getLastResult(): ?CategoryMerchandisingResult
+    public function getLastResult(): CategoryMerchandisingResult
```
Comment (olsi-qose) [inline Model/Service/Recommendation/StateAwareCategoryService.php:250]: This will result in unhandled exceptions in case `getPersonalisationResult()` fails. `lastResult` will never have a value and other interceptors rely on that. 

Example:
`main.CRITICAL: TypeError: Return value of Nosto\Cmp\Model\Service\Recommendation\StateAwareCategoryService::getLastResult() must be an instance of Nosto\Result\Graphql\Recommendation\CategoryMerchandisingResult, null returned in /magento2/vendor/nosto/module-nostotagging-cmp/Model/Service/Recommendation/StateAwareCategoryService.php:252`

File: Plugin/Framework/Search/Request/AbstractHandler.php [hunk 2]
```diff
@@ -132,12 +132,15 @@ public function __construct(
 
     /**
      * @param array $requestData
+     * @param bool $doLog
      * @return void
      * @noinspection PhpRedundantCatchClauseInspection
      */
-    public function handle(array &$requestData)
+    public function handle(array &$requestData, $doLog)
```
Comment (olsi-qose) [inline Plugin/Framework/Search/Request/AbstractHandler.php:139]: Why is this added here? Having a variable to control logging will add more confusion and make debugging harder.

File: Model/Service/Recommendation/StateAwareCategoryService.php (hunk 1 above)
Comment (olsi-qose) [reply Model/Service/Recommendation/StateAwareCategoryService.php:250]: You can add a clause to check that the result is instance of `CategoryMerchandisingResult `. But the aftermath will depend on each case.

File: Plugin/Framework/Search/Request/AbstractHandler.php [hunk 3]
```diff
@@ -132,12 +132,15 @@ public function __construct(
...
     {
-        $this->trace('Using %s as search engine', [$this->searchEngineHelper->getCurrentEngine()]);
+        if ($doLog) {
```
Comment (olsi-qose) [inline Plugin/Framework/Search/Request/AbstractHandler.php:141]: Remove.

File: Plugin/Framework/Search/Request/AbstractHandler.php [hunk 4]
```diff
@@ -152,7 +155,9 @@ public function handle(array &$requestData)
             );
             //In case CM category is not configured in nosto
             if ($productIds == null || empty($productIds)) {
-                $this->trace('Nosto did not return products for the request', [], $requestData);
+                if ($doLog) {
```
Comment (olsi-qose) [inline Plugin/Framework/Search/Request/AbstractHandler.php:158]: Remove.

File: Plugin/Framework/Search/Request/RequestCleaner.php [hunk 5]
```diff
@@ -89,17 +89,19 @@ public function __construct(
     // phpcs:ignore Generic.CodeAnalysis.UnusedFunctionParameter
     public function afterClean(Cleaner $cleaner, array $requestData)
     {
+        $doLog = true;
```
Comment (olsi-qose) [inline Plugin/Framework/Search/Request/RequestCleaner.php:92]: Remove,

**Review by olsi-qose — CHANGES_REQUESTED** [review]

File: Model/Service/Recommendation/StateAwareCategoryService.php (hunk 1 above)
Comment (dairbuirabass) [reply Model/Service/Recommendation/StateAwareCategoryService.php:250]: My idea was to initialise empty `CategoryMerchandisingResult` even if there is none. This way null checks are not necessary in interceptors.

File: Plugin/Framework/Search/Request/AbstractHandler.php (hunk 2 above)
Comment (dairbuirabass) [reply Plugin/Framework/Search/Request/AbstractHandler.php:139]: Since I am calling handle function no matter wether CMP is enabled or not I added logging argument to differentiate between where it actually logs something useful and where it just needs to be executed for initialising empty `CategoryMerchandisingResult` object
